test/core/event_engine/test_suite/tests/dns_test.cc

```diff
@@ -25,7 +63,529 @@ void InitDNSTests() {}
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
-class EventEngineDNSTest : public EventEngineTest {};
+namespace {
+
+// TODO(yijiem): build a portable solution for Windows
+constexpr char kDNSTestRecordGroupsYamlPath[] =
+    "test/core/event_engine/test_suite/tests/dns_test_record_groups.yaml";
+constexpr char kHealthCheckRecordName[] =
+    "health-check-local-dns-server-is-alive.resolver-tests.grpctestingexp";
+
+// You know where this comes from.
+template <typename T>
+void WaitForSingleOwner(std::shared_ptr<T>&& object) {
+  while (object.use_count() > 1) {
+    GRPC_LOG_EVERY_N_SEC(2, GPR_INFO, "engine.use_count() = %ld",
+                         object.use_count());
+    absl::SleepFor(absl::Milliseconds(100));
+  }
+}
+
+}  // namespace
+
+class EventEngineDNSTest : public EventEngineTest {
+ public:
+  static void SetUpTestSuite() {
+    // Invoke bazel's executeable links to the .sh and .py scripts (don't use
+    // the .sh and .py suffixes) to make sure that we're using bazel's test
+    // environment.
+    std::string kPythonWrapper = "tools/distrib/python_wrapper";
+    std::string kDNSServerPath = "test/cpp/naming/utils/dns_server";
+    std::string kDNSResolverPath = "test/cpp/naming/utils/dns_resolver";
+    std::string kTCPConnectPath = "test/cpp/naming/utils/tcp_connect";
+    // HACK: Hyrum's law.
+    if (!grpc_core::GetEnv("TEST_SRCDIR").has_value()) {
+      // Invoke the .sh and .py scripts directly where they are in source code.
+      kPythonWrapper += ".sh";
+      kDNSServerPath += ".py";
+      kDNSResolverPath += ".py";
+      kTCPConnectPath += ".py";
+    }
+    // 1. launch dns_server
+    int port = grpc_pick_unused_port_or_die();
+    ASSERT_NE(port, 0)
+        << "pick unused port failed, maybe the port server is not running? "
+           "Start it with tools/run_tests/start_port_server.py";
+    // <path to python wrapper> <path to dns_server.py> -p <port> -r <path to
+    // records config>
+    _dns_server.server_process = new grpc::SubProcess(
+        {kPythonWrapper, kDNSServerPath, "-p", std::to_string(port), "-r",
+         kDNSTestRecordGroupsYamlPath});
+    // TODO(yijiem): no way to check whether it fails or not
```

Comment:
That may be a good reason to do this setup in another tool.

One option could be doing a DNS query for some hard-coded entry in the resolver.

Another might be adding platform-specific ways to check if a process is alive to `grpc_core::SubProcess`. The pid is available, but not exposed.